### ISSUE TITLE
Invalid GloVe results

### ISSUE THREAD

**Issue by czyzby** [eclipse/deeplearning4j#3462]
#### Issue Description

GloVe yields unexpected results when running through `org.deeplearning4j.examples.nlp.glove.GloVeExample`.

A) It might return `NaN` as word similarity (this seems to be seed-dependent):
```
o.d.e.n.g.GloVeExample - Day/night similarity: NaN
o.d.e.n.g.GloVeExample - Nearest words to 'day': [i, do, to, nt, the, that, s, he, you, is]
```

B) Even when reporting similarities that are actual numbers, the results are way off:
```
o.d.e.n.g.GloVeExample - Day/night similarity: 0.06969143450260162
o.d.e.n.g.GloVeExample - Nearest words to 'day': [next, yesterday, a, season, place, one, who, program, best, house]
```

For comparison, these are word2vec results using the same data:
```
o.d.e.n.w.Word2VecRawTextExample - Day/night similarity: 0.7510488033294678
o.d.e.n.w.Word2VecRawTextExample - Closest Words:
10 Words closest to 'day': [night, week, year, game, season, time, group, office, director, show]
```

#### Version Information

Deeplearning4J 0.8.0.

**Comment by daanvdn** [comment]
@raver119 , I've been testing this on dl4j version 1.0.0-alpha and 1.0.0-beta and I'm also experiencing this very same issue (whether it's through `org.deeplearning4j.examples.nlp.glove.GloVeExample`  or any other code that uses `org.deeplearning4j.models.glove.Glove`). I've tried it on 3 different machines (windows 7 vs 1 Windows Server 2016 vs Centos 6.9) and they all have the same problem.  Also tried changing the DataType but that doesn't help. 
Is there any workaround for this issue? If not, does this mean that effectively the dl4j implementation of Glove cannot be used at the moment? 
I saw there is a related/duplicate ticket that's been closed. Mentioning it here for reference's sake:  https://github.com/deeplearning4j/deeplearning4j/issues/1622
